cardano-cli/cardano-cli.cabal

```diff
@@ -174,6 +174,17 @@ executable cardano-cli
                       , optparse-applicative-fork
                       , transformers-except
 
+library test-lib
+  import:               project-config
```

Comment:
https://github.com/input-output-hk/cardano-node/pull/5200#discussion_r1190006088

Reply:
I merged those test components again to unblock this PR.

The tests need some cleanup in general.

It's unclear which files are test inputs, which are golden files and which are unused, tests need to be manually registered in `Main` and consistent naming means it's difficult to run tests selectively.

But cleanup can wait until a future PR.

Reply:
https://github.com/input-output-hk/cardano-node/pull/5200#discussion_r1197226656